alacritty/src/event.rs

```diff
@@ -456,13 +455,22 @@ impl<N: Notify> Processor<N> {
                 },
                 Event::ConfigReload(path) => {
                     processor.ctx.message_buffer.remove_target(LOG_TARGET_CONFIG);
-                    processor.ctx.resize_pending.message_buffer = Some(());
+                    processor.ctx.display_update_pending.message_buffer = Some(());
 
                     if let Ok(config) = config::reload_from(&path) {
+                        if processor.ctx.font.size == processor.config.font.size {
+                            processor.ctx.display_update_pending.font = Some(FontUpdate::Reset);
+                        }
                         processor.ctx.terminal.update_config(&config);
 
-                        if *processor.ctx.font_size == processor.config.font.size {
-                            processor.ctx.resize_pending.font_size = Some(FontResize::Reset);
+                        // use copy of new config with font size overridden to match the current
+                        // font size in order to determine if any font value other than size has
+                        // changed
+                        let mut font_copy = config.font.clone();
+                        font_copy.size = processor.ctx.font.size;
+
+                        if *processor.ctx.font != font_copy {
+                            processor.ctx.display_update_pending.font = Some(FontUpdate::Reset);
                         }
```

Comment:
What if the font size has changed, but the current font size is not the default font size? We don't want to reset to the config's font then, because that has the incorrect size.

Reply:
I guess I was assuming that if something other than the font size changed we would want the font reset regardless of whether the font size has changed dynamically since starting. But I can see how some might find that behavior surprising.

I think if we pass a `Font` as a parameter to the `Reset` enum in this case we could pass in a copy with the current size overridden by the value on `processor.ctx.font` to ensure that the `Display`'s font doesn't change when its family or style changes.

Based on your reaction to the `config.font.clone()` below, I suspect you might not be comfortable with that in which case I'm not sure how we can change any other part of the font when the size has been dynamically changed.

Reply:
Why would this be a problem? Cloning the font to send it as an event shouldn't be a massive problem.

Reply:
I guess in my ignorance I'm just not quite at the point where I can adequately interpret what was wrong that prompted https://github.com/jwilm/alacritty/pull/2836/files#r338834447; but that is kind of moot now as I am about to push an update that should address all the feedback so far.